### PR TITLE
Add AudioQnA example

### PR DESCRIPTION
**Pull request by Spycsh**
## Type of Change

feature

## Description

add AudioQnA example

Already done:

- [x] ASR/TTS as microservices based on Whisper and GPT-SOVITS
- [x] TTS optimization, bf16 and libiomp and tcmalloc
- [x] allow users to upload default voice for zero-shot voice cloning
- [x] ChatQnA migration for LLM part
- [x] add tests
- [x] add draft README

TODO:

- [x] add ui (@WenjiaoYue)
- [x] enable CI (@chensuyue @XuehaoSun )

Usage

```
# Start ASR service
docker run -d -e http_proxy=$http_proxy -e https_proxy=$https_proxy -p 8008:8008 intel/gen-ai-examples:audioqna-asr

# Test ASR
wget https://github.com/intel/intel-extension-for-transformers/raw/main/intel_extension_for_transformers/neural_chat/assets/audio/sample.wav
http_proxy= curl -F 'file=@sample.wav' http://localhost:8008/asr

# Start TTS service
# Predownload local models and mapped in
git clone https://huggingface.co/lj1995/GPT-SoVITS pretrained_tts_models
docker run -d -v ./pretrained_tts_models:/GPT-SoVITS/GPT_SoVITS/pretrained_models -e http_proxy=$http_proxy -e https_proxy=$https_proxy -p 9880:9880 intel/gen-ai-examples:audioqna-tts --bf16

# Upload reference audio
http_proxy= curl --location 'localhost:9880/upload_as_default' \
--form 'default_refer_file=@"sample.wav"' \
--form 'default_refer_text="Who is Pat Gelsinger?"' \
--form 'default_refer_language="en"'

# Test TTS
http_proxy= curl --location 'localhost:9880' \
--header 'Content-Type: application/json' \
--data '{
    "text": "You can have a look, but you should not touch this item.",
    "text_language": "en"
}' \
--output output.wav
```

## How has this PR been tested?

three tests under `tests/`

## Dependency Change?

Check Dockerfile

### CODE REVIEW

File: AudioQnA/langchain/docker/qna-app/app/NeuralChatEndpoint.py
Comment (lvliang-intel) [inline AudioQnA/langchain/docker/qna-app/app/NeuralChatEndpoint.py:1]: Please remove this file.

File: AudioQnA/audio/docker/asr/asr_server.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from asr import AudioSpeechRecognition
+from fastapi import FastAPI, File, UploadFile
+from fastapi.responses import Response
+from pydub import AudioSegment
+
+app = FastAPI()
+asr = None
+
+
+@app.get("/health")
```
Comment (lvliang-intel) [inline AudioQnA/audio/docker/asr/asr_server.py:31]: /v1/health

File: AudioQnA/audio/docker/asr/asr_server.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+async def health() -> Response:
+    """Health check."""
+    return Response(status_code=200)
+
+
+@app.post("/asr")
```
Comment (lvliang-intel) [inline AudioQnA/audio/docker/asr/asr_server.py:37]: should be compatible with openai "/v1/audio/transcriptions"

File: AudioQnA/audio/docker/tts/tts_server.py [hunk 3]
```diff
@@ -0,0 +1,716 @@
...
+        json_post_raw.get("refer_wav_path"), json_post_raw.get("prompt_text"), json_post_raw.get("prompt_language")
+    )
+
+
+@app.get("/change_refer")
+async def change_refer(refer_wav_path: str = None, prompt_text: str = None, prompt_language: str = None):
+    return handle_change(refer_wav_path, prompt_text, prompt_language)
+
+
+@app.post("/")
```
Comment (lvliang-intel) [inline AudioQnA/audio/docker/tts/tts_server.py:667]: should be compatible with openai api "/v1/audio/speech"

File: AudioQnA/audioqna.jpg
Comment (lvliang-intel) [inline AudioQnA/audioqna.jpg:1]: Please upload the pic to https://i.imgur.com and just put the image link in the readme.

File: AudioQnA/tests/test_langchain_inference.sh [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    fi
+}
+
+function docker_stop() {
+    local container_name=$1
+    cid=$(docker ps -aq --filter "name=$container_name")
+    if [[ ! -z "$cid" ]]; then docker stop $cid && docker rm $cid; fi
+}
+
+function main() {
```
Comment (chensuyue) [inline AudioQnA/tests/test_langchain_inference.sh:103]: Did you update this test case? This seems to be a copy from ChatQnA.

Comment (Spycsh) [reply AudioQnA/tests/test_langchain_inference.sh:103]: This is just the same copy from ChatQnA

Comment (chensuyue) [reply AudioQnA/tests/test_langchain_inference.sh:103]: Please update the test scripts for this case, I will add the pipeline to trigger the test.

Comment (Spycsh) [reply AudioQnA/tests/test_langchain_inference.sh:103]: Will do later!

**Pull request merged**